### DISCUSSION TITLE
How to visualize CNN filters, and feature maps?

### DISCUSSION THREAD

**Discussion by elcolie** [Lightning-AI/pytorch-lightning#7803] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
I am studying the CNN and found that pytorch lightning is very useful. I can see `loss` in real time in tensorboard. But I am not sure how to find the answer for following questions
1. How to Visualize Filters
1. How to Visualize Feature Maps

Any short example would be appreciated
Thank you

**Comment by Borda** [comment]
hi, there, mind check this blog [Looking Inside The Blackbox — How To Trick A Neural Network](https://towardsdatascience.com/peering-inside-the-blackbox-how-to-trick-a-neural-network-757c90a88a73)
you can also find the implementation in Bolts:
https://github.com/PyTorchLightning/lightning-bolts/blob/2163626e47952207e07a53e082e692a711de5afc/pl_bolts/callbacks/vision/confused_logit.py#L20-L167
